glossary.Rmd

```diff
@@ -0,0 +1,163 @@
+
+```{r setup, include = FALSE}
+source("setup.R")
+library("dplyr")
+```
+
+# Glossary
+
+This glossary contains all the vocabulary necessary to work with tidy evaluation and, more generally, with [expressions](#glossary-expr). The definitions in rlang are generally consistent with base R. When they differ, both definitions are presented so you can navigate between these two worlds more easily.
+
+
+## Data structures
+
+### TODO Data mask {#glossary-data-mask}
+
+### Expression {#glossary-expr}
+
+An expression is a piece of R code that describes a value:
+
+```{r}
+12
+```
+
+or a computation:
+
+```{r}
+12 / 3
+```
+
+Expressions are normally transient. They are executed ([evaluated](#glossary-eval)) when you source a file or call a function and you only see their final value or their side effects (such as console output). However in R you can suspend the normal evaluation of expressions with the [quotation](#glossary-quotation) mechanism:
+
+```{r}
+# Evaluated expression
+12 / 3
+
+# Quoted expression
+expr(12 / 3)
+```
+
+The technical definition of expressions is any R object that is created by [parsing](#glossary-parse) R code:
+
+* [Constants](#glossary-constant-symbolic) like `NULL`, `1`, `"foo"`, `TRUE`, `NA`, etc.
+* [Symbols](#glossary-sym) like `height` or `weight`
+* [Calls](#glossary-call) like `c()` or `list()`
+
+Unlike constants, symbols and calls are [symbolic](#glossary-constant-symbolic) objects: their [value](#glossary-eval) depends on the [environment](#glossary-env).
+
+
+### Expression (base)
+
+In base R, "expression" refers to a special type of vectors that contains quoted expressions in the rlang sense:
+
+```{r}
+base::expression(key <- "foo", toupper(key))
+```
+
+You'll most likely encounter this rare data structure as the return value of `base::parse()`:
+
+```{r}
+code <- "key <- 'foo'; toupper(key)"
+parse(text = code)
+```
+
+The only advantage of expression vectors compared to lists is that they include **source references**. The most apparent manifestation of source references is that expression vectors are printed with whitespace and comments preserved:
```

Comment:
I think you can simplify the second sentence here:

Such expression vectors with source references are printed with…